src/main/java/leo/storage/Task.java

```diff
@@ -0,0 +1,105 @@
+package leo.storage;
+
+import leo.leoexception.IncorrectMarkException;
+import leo.leoexception.IncorrectUnmarkException;
+import leo.ui.Ui;
+
+/**
+ * Represents a Task. A <code>Task</code> object contains the String description.
+ */
+public class Task {
```

Comment:
Perhaps `task` could be an abstract class as well? Making the other methods abstract will allow you to ensure that the implementation in the child classes will be used.